buffer/src/main/java/io/netty/buffer/api/BufferClosedException.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+/**
+ * An exception thrown when an operation is attempted on a {@link Buffer} when it has been closed.
+ */
+public final class BufferClosedException extends UnsupportedOperationException {
```

Comment:
Why `UnsupportedOperationException` rather than `IllegalStateException`?

Reply:
This matches the exceptions thrown from the JDK `ByteBuffer`.

Reply:
`ByteBuffer` doesn't have a `close()` method. Which method in `ByteBuffer` does throw an `UnsupportedOperationException` when the buffer is in an unexpected state? A read-only buffer throws UOE but it's because write operations are 'not supported'. In this case, 'close' operation is not 'unsupported' but it's just a user attempted to access a close buffer (i.e. a buffer in an 'bad state').

Reply:
Hmm… that's a good point.